src/app/(profile)/profiles/[username]/page.tsx

```diff
@@ -44,11 +47,22 @@ const UserProfilePage = async ({
   params: { username: string }
 }) => {
   const selectedProfile = await getProfileByGithubUsername(params.username)
+  const session = await getServerSession(authOptions)
 
   if (!selectedProfile) {
     redirect(AppRoutes.profiles)
   }
 
+  if (!session?.user) {
+    redirect(AppRoutes.profiles)
+  }
+
+  try {
+    await countProfileView(selectedProfile.id, session.user.id)
```

Comment:
**nit:** Let's add return here, because as I understand it returns updated profile?

Reply:
Yes, it can return updated profile data. I skipped it because its not passed anywhere. Data presented to user in 'myprofile' is loaded from db.  I will add this if necessary